### PR TITLE
Add Smithy protocol test client pkg generation

### PR DESCRIPTION
**Pull request by jasdel**
Adds generation of smithy core's protocol test client packages.

### CODE REVIEW

File: codegen/protocol-test-codegen/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *     or in the "license" file accompanying this file. This file is distributed
+ *     on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *     express or implied. See the License for the specific language governing
+ *     permissions and limitations under the License.
+ */
+
+import software.amazon.smithy.gradle.tasks.SmithyBuild
+
+plugins {
+    id("software.amazon.smithy") version "0.4.3"
```
Comment (skotambkar) [inline codegen/protocol-test-codegen/build.gradle.kts:19]: is this version correct?

File: codegen/sdk-codegen/aws-models/lex-runtime-service.2016-11-28.json [hunk 2]
```diff
@@ -1,29 +1,29 @@
 {
-  "smithy": "1.0",
+  "smithy": "1",
```
Comment (skotambkar) [inline codegen/sdk-codegen/aws-models/lex-runtime-service.2016-11-28.json:2]: I think smithy usually uses 1.0, any reason for changing to 1?

File: codegen/protocol-test-codegen/build.gradle.kts (hunk 1 above)
Comment (jasdel) [reply codegen/protocol-test-codegen/build.gradle.kts:19]: probably not, this was copied from JS will take a look

Comment (skotambkar) [reply codegen/protocol-test-codegen/build.gradle.kts:19]: should probably be 1.0?

File: codegen/sdk-codegen/aws-models/lex-runtime-service.2016-11-28.json (hunk 2 above)
Comment (jasdel) [reply codegen/sdk-codegen/aws-models/lex-runtime-service.2016-11-28.json:2]: This was failing if 1.0 was used, and hand to change it to 1. will go back and verify

**Pull request merged**